diff --git a/make/build-all.sh b/make/build-all.sh
index 035f9b8..edd19c0 100644
--- a/make/build-all.sh
+++ b/make/build-all.sh
@@ -61,9 +61,9 @@ mixed_path() {
 }
 
 get_scm_type() {
-    if [ -d .hg ]; then
+    if [ -e .hg ]; then
         echo "HG"
-    elif [ -d .git ]; then
+    elif [ -e .git ]; then
         echo "GIT"
     else 
         echo "Error: unrecognized repository, it must be Git or Mercurial" >&2
